### DISCUSSION TITLE
Using nifty file as input for my MAP application

### DISCUSSION THREAD

**Discussion by ArpanGyawali** [Project-MONAI/monai-deploy-app-sdk#462] [General]
I have created MAP for taking dicom data as input and get the segmentation for skull stripping task following the spleen seg inference application. 
We have more data in nifty format than in dicom so I wanted to create a additional application or combine with the previous that takes nifty as well. 
I tried using Nii_file_loader_operator but it returns numpy array(np.ndarray) and when i tried to give this output to skull_strip_operator(similar to spleen_seg_operator), as it takes monai.core.Image as input. So this raises an error

So i edited the Nii_file_loader_operator by just changing the type explicitly by
`op_output.set(Image(image_np))`

But this returns Image without any metadata headet so i got the error like
**KeyError: 'row_pixel_spacing'** thrown by LoadImaged pretransform.

**What should i do so that my LoadImaged pretransform takes nifty image from my disk or numpy image returned by nifty_loader from InMemory.**

**Comment by MMelQin** [comment]
@ArpanGyawali A Numpy array as the result of an image loader is never enough in the case of inference. The loader need to provide both the Numpy array as well as the metadata. Further, to fit what's expected by the built-in inference operators, an App SDK Image object is required, which really is just numpy array with a list of expected metadata.

A more efficient implementation is actually do what's done in MONAI core, i.e. passing the file path to the `LoadImaged `transform instead of using the custom in_mem image reader. The one thing is that this requires extending and creating a custom inference operator.

**Reply by ArpanGyawali** [reply]
thank you @MMelQin 
But I havent used built in inference operator.
For creating the skull strip application I have used the following operator (version: sdk 0.5.1)

    Dicom Loader
    Dicom Series Selector
    Dicom Series to Volume
    Custom SkullStrip operator (similar to spleen seg operator)
    Dicom Writer
    and used the model that i got after training custom data in monai label.

So according to you, I can create a custom inference operator where the pretransform LoadImaged can take the nifty file path directly instead of using any other nifty loader operator?
How can I do this? Is there any references?
Any help would be deeply appreciated. Thank you
